### PR TITLE
Imgix resolver

### LINKED ISSUE #31: imgix.com resolver for image thumbnails
imgix.com offers media CDN that supports automatic thumbnailing. We could have a resolver that would automagically resolve from attribute with image URL to final URL that has thumbnail.

### PR DESCRIPTION
**Pull request by mat-sop**
resolves #31

### CODE REVIEW

File: GUIDE.md [hunk 1]
```diff
@@ -446,6 +446,83 @@ app = GraphQL(
 ```
 
 
+## imgix query params resolver
+
+`get_query_params_resolver` returns a preconfigured resolver that takes URL string and passed arguments to generate a URL with arguments as query params. It can be used to add [rendering options](https://docs.imgix.com/apis/rendering) to [imgix.com](https://imgix.com) image URL.
+
+### Function arguments:
```
Comment (rafalp) [inline GUIDE.md:453]:
```suggestion

### Function arguments:
```

File: GUIDE.md [hunk 2]
```diff
@@ -446,6 +446,83 @@ app = GraphQL(
...
+
+- `get_url`: a `str` or `Callable` which returns `str`. If `get_url` is a `str` then the resolver will split it by `.` and use substrings as keys to get value from `obj` dict, e.g. with `get_url` set to `"imageData.url"` the resolver will use `obj["imageData"]["url"]` as URL string. If `get_url` is a callable, then resolver will call it with `obj`, `info` and `**kwargs` and use result as URL string.
+- `extra_params`: an optional `dict` of query params to be added to the URL string. These can be overridden by kwargs passed to the resolver.
+- `get_params`: an optional `Callable` to be called on passed `**kwargs` before they are added to the URL string.
+- `serialize_url`: an optional `Callable` to be called on URL string with query params already added. Result is returned directly by the resolver.
+
+### Example with `insert_field`
```
Comment (rafalp) [inline GUIDE.md:460]:
```suggestion

### Example with `insert_field`
```

File: GUIDE.md [hunk 3]
```diff
@@ -762,6 +839,20 @@ def get_sub_schema(self, schema_id: int) -> GraphQLSchema:
...
 
+### `insert_field`
+
+```python
+def insert_field(self, type_name: str, field_str: str):
+```
+
+Inserts field into all schemas with given `type_name`. The field is automatically delayed - excluded from queries run by `root_resolver` against the remote GraphQL APIs.
+
+#### Required arguments
```
Comment (rafalp) [inline GUIDE.md:850]:
```suggestion

#### Required arguments
```

File: ariadne_graphql_proxy/contrib/imgix/query_params_resolver.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+from graphql import GraphQLResolveInfo
+
+
+def get_attribute_value(
+    obj: Any, info: GraphQLResolveInfo, attribute_str: str, **kwargs
+) -> Any:
+    value = obj
+    for attr in attribute_str.split("."):
+        value = value.get(attr)
```
Comment (rafalp) [inline ariadne_graphql_proxy/contrib/imgix/query_params_resolver.py:13]: We should have try/except test and do `getattr(value, attr, None)` in except if `get()` fails. This is to support other types than dicts.

**Review by rafalp — CHANGES_REQUESTED** [review]

File: ariadne_graphql_proxy/contrib/imgix/query_params_resolver.py (hunk 4 above)
Comment (mat-sop) [reply ariadne_graphql_proxy/contrib/imgix/query_params_resolver.py]: Done, I also added tests with non dict objects and mentioned them in `GUIDE.md`

**Review by rafalp — APPROVED** [review]

**Pull request merged**